net/libfab/libfab.c

```diff
@@ -61,54 +61,85 @@
 #include "libfab_internal.h"
 #include <errno.h>
 
+#define LIBFAB_VERSION FI_VERSION(FI_MAJOR_VERSION,FI_MINOR_VERSION)
 
-/** Network transfer machine */
-struct transfer_ma {
-	/**
-	* Generic transfer machine with buffer queues, etc. 
-	*/
-	struct m0_net_transfer_mc *t_ma;
+static char *providers[] = { "verbs", "tcp", "sockets" };
 
-	/**
-	* TODO: Is poller thread required ?
-	*/
-};
+static int libfab_dom_find(const char *name, struct fi_info *fi_list);
+static int libfab_ep_res_init(struct m0_fab__dom_param *fab,
+			      struct m0_fab__ep_param *ep);
+static struct m0_fab__ep_param *libfab_ep_net(struct m0_net_end_point *net);
+static int libfab_ep_find(const char *name, struct m0_net_transfer_mc *tm, 
+			  struct m0_net_end_point **epp);
+static void libfab_ep_param_free(struct m0_fab__ep_param *ep);
+static void libfab_ep_res_free(struct m0_fab__ep_res *ep_res);
+
+/* Find and select the domain from the list */
+static int libfab_dom_find(const char *name, struct fi_info *fi_list)
+{
+	struct fi_info *fab = fi_list;
+	int             rc = 1;
+
+	M0_ENTRY();
+
+	if (name == NULL)
+		return M0_RC(0);
+
+	while ((fab != NULL) && (rc != 0)) {
+		rc = strcmp(name, fab->domain_attr->name);
+		if (rc == 0)
+			fi_list = fab;
```

Comment:
Why do you update the variable `fi_list`?

This assignment has no effect outside this function.